master/pkg/model/task_container_defaults.go

```diff
@@ -134,8 +134,19 @@ func (c *TaskContainerDefaultsConfig) MergeIntoExpConfig(config *expconf.Experim
 	bindMounts := c.BindMounts.ToExpconf()
 	config.RawBindMounts = schemas.Merge(config.RawBindMounts, bindMounts)
 
+	configRawSlurmConfig := config.RawSlurmConfig
 	config.RawSlurmConfig = schemas.Merge(config.RawSlurmConfig, &c.Slurm)
+	if configRawSlurmConfig != nil {
+		config.RawSlurmConfig.RawSbatchArgs = append(
+			c.Slurm.SbatchArgs(), configRawSlurmConfig.SbatchArgs()...)
+	}
+
+	configRawPbsConfig := config.RawPbsConfig
 	config.RawPbsConfig = schemas.Merge(config.RawPbsConfig, &c.Pbs)
+	if configRawPbsConfig != nil {
+		config.RawPbsConfig.RawSbatchArgs = append(
+			c.Pbs.SbatchArgs(), configRawPbsConfig.SbatchArgs()...)
+	}
 }
```

Comment:
do this work? seems like avoiding `schemas.Merge` all together avoids confusion, to me. it is much easier for me to read and understand the merge behavior like this.
```suggestion
	slurmConfig := config.SlurmConfig()
	slurmConfig.SetSbatchArgs(append(c.Slurm.SbatchArgs(), slurmConfig.SbatchArgs()...))
	config.SetSlurmConfig(slurmConfig)
```

Reply:
| do this work? seems like avoiding schemas.Merge all together avoids confusion, to me. it is much easier for me to read and understand the merge behavior like this.

I don't think it does because you may or may not have a gpu_type or slots_per_node to inherit.  Even c.Slurm and c.Pbs might be null which makes the necessary expressions really complicated.  Thus I left the merge to handle all of those cases, and only apply the merging logic to the batch args that needed it.

Reply:
yeah, i didn't test what i wrote so it isn't exact, cleaned up below, still seems clearer. idk, personal preference
```suggestion
	sbatchArgs := append(c.Slurm.SbatchArgs(), config.SlurmConfig().SbatchArgs()...)
	slurmConfig := schemas.Merge(config.RawSlurmConfig, &c.Slurm)
	slurmConfig.SetSbatchArgs(sbatchArgs)
	config.RawSlurmConfig = slurmConfig

	qsubArgs := append(c.Pbs.SbatchArgs(), config.PbsConfig().SbatchArgs()...)
	pbsConfig := schemas.Merge(config.RawPbsConfig, &c.Pbs)
	pbsConfig.SetSbatchArgs(qsubArgs)
	config.RawPbsConfig = pbsConfig
```

Reply:
or clearer https://go.dev/play/p/bwxP0gnv93n (shows `var xs []int` is indeed nil)